Sources/Recycling/GridViews/Ads/Model/AdsGridImageCache.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import Foundation
+
+private class WeakImage {
+    private(set) weak var value: UIImage?
+
+    init(value: UIImage) {
+        self.value = value
+    }
+}
+
+class AdsGridImageCache {
+    private var cache: [String: WeakImage] = [:]
```

Comment:
Might be worth using `NSCache` instead of dictionary here. Then you can set some count and cost limit and objects will be automatically removed from the cache if the device is running low on memory.